src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs

```diff
@@ -134,5 +134,8 @@ public interface INamedTypeSymbol : ITypeSymbol
         /// If false, the symbol does not contain extension methods. 
         /// </summary>
         bool MightContainExtensionMethods { get; }
+
+        ImmutableArray<ITypeSymbol> TupleElementTypes { get; }
+        ImmutableArray<string> TupleElementNames { get; }
```

Comment:
We already have test for these methods in VB (always returns nothing) and positive case in C# (returns proper arrays on tuple types). 
I'll add negative case for C# (returns default arrays on non-tuple types).